src/RouterError.ts

```diff
@@ -0,0 +1,15 @@
+import { RouterErrorCode } from "./types"
+
+/**
+ * Represents a failure to route an event.
+ */
+export default class RouterError extends Error {
+    public isRequestParseError = true
```

Comment:
Is this required?

Reply:
yee my bad

Reply:
i changed it to `isRouterError`. The reason being is then you can also check `error.isRouterError`, as well as `err instanceof RouterError`.